web/src/app/services/loi/loi.service.ts

```diff
@@ -39,24 +41,35 @@ export class LocationOfInterestService {
   private selectedLocationOfInterest$: Observable<LocationOfInterest>;
 
   constructor(
+    private authService: AuthService,
     private dataStore: DataStoreService,
     private surveyService: SurveyService
   ) {
-    this.lois$ = surveyService
-      .getActiveSurvey$()
+    this.lois$ = this.authService
+      .getUser$()
       .pipe(
-        switchMap(survey =>
-          survey.isUnsavedNew()
-            ? of(List<LocationOfInterest>())
-            : dataStore.lois$(survey)
+        switchMap(user =>
+          this.surveyService
+            .getActiveSurvey$()
+            .pipe(
+              switchMap(survey =>
+                survey.isUnsavedNew()
+                  ? of(List<LocationOfInterest>())
+                  : dataStore.lois$(survey, user)
+              )
+            )
         )
       );
 
     this.selectedLocationOfInterest$ = this.selectedLocationOfInterestId$.pipe(
       switchMap(loiId =>
-        surveyService.getActiveSurvey$().pipe(
-          switchMap(survey => dataStore.lois$(survey)),
-          map(lois => lois.find(loi => loi.id === loiId)!)
+        this.authService.getUser$().pipe(
```

Comment:
Can we rename this `selectedLoi$` while we're at it for consistency with other field and var names?